iceoryx_posh/test/moduletests/test_roudi_iceoryx_roudi_app.cpp

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_posh/internal/log/posh_logging.hpp"
+#include "iceoryx_posh/roudi/iceoryx_roudi_app.hpp"
+#include "iceoryx_posh/roudi/roudi_cmd_line_parser_config_file_option.hpp"
+#include "iceoryx_posh/roudi/roudi_config_toml_file_provider.hpp"
+
+#include "test.hpp"
+
+using namespace ::testing;
+using ::testing::Return;
+
+using iox::roudi::IceOryxRouDiApp;
+using namespace iox::config;
+
+namespace iox
+{
+namespace test
+{
+class IceoryxRoudiApp_Child : public IceOryxRouDiApp
+{
+  public:
+    IceoryxRoudiApp_Child(const config::CmdLineArgs_t& cmdLineArgs, const RouDiConfig_t& roudiConfig)
+        : IceOryxRouDiApp(cmdLineArgs, roudiConfig)
+    {
+    }
+
+    bool getVariableRun()
+    {
+        return m_run;
+    }
+
+    iox::log::LogLevel getLogLevel()
+    {
+        return m_logLevel;
+    }
+
+    roudi::MonitoringMode getMonitoringMode()
+    {
+        return m_monitoringMode;
+    }
+
+    RouDiConfig_t getConfig()
+    {
+        return m_config;
+    }
+
+    void setVariableRun(bool condition)
+    {
+        m_run = condition;
+    }
+
+    uint8_t callRun() noexcept
+    {
+        return run();
+    }
+};
+
+/// @brief Test goal: This file tests class roudi app
+class IceoryxRoudiApp_test : public Test
+{
+  public:
+    CmdLineParserConfigFileOption cmdLineParser;
+
+    void TearDown()
+    {
+        // Reset optind to be able to parse again
+        optind = 0;
```

Comment:
You need to add 
```cpp
#include "iceoryx_utils/platform/getopt.hpp"
```
in this file so that optind can be found on windows. We added there a definition for the windows build. See https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_utils/platform/win/include/iceoryx_utils/platform/getopt.hpp
optind is a unix specific construct and not known to windows, later we need to write a wrapper construct for optind but that is not part of this PR.

Reply:
Look like it's done.